### PR TITLE
RNO-G surface trigger

### PR DESCRIPTION
**Pull request by lpyras**
This PR includes a parametrisation of the Schottky diode used for RNO-G. The trigger path includes a bandpass filter from 80-180 MHz and an attenuator of -10dB. The sensitivity of the FPGA towards the threshold and signal is not implemented yet. Currently RNO-G is running on a 1/3 coincidence for upward facing LPDA. This might change in the future.

The parameters are obtained with a linear fit of max(V_in)^2 versus max(abs(V_out)), with measurements done at DESY.

cr_trace_393_29_io_T
cr_trace_393_29_io_T22
example_fit_1V
.

### CODE REVIEW

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (anelles) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:38]: I would suggest not to use such a generic name in overwriting standard functionality. More like linear_function etc.

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (anelles) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:27]: Can you rename to temperature? This has me very comfused as this is the standard abbreviation of temporary variable.

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (anelles) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:119]: See above

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (anelles) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:158]: remove

File: NuRadioReco/framework/trigger.py
Comment (anelles) [inline NuRadioReco/framework/trigger.py:368]: see below

**Review by anelles — CHANGES_REQUESTED** [review]
Looks good to me. Please address the detailed comments. And could you also add the scirpts that make the calibration plots (which you only added as figure file?)

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (sjoerd-bouma) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:19]: docstring nitpicking - can you add an empty line after the first line here? Docstrings should start off
```
short single line description

(optional) longer description
```

File: NuRadioReco/framework/trigger.py
Comment (sjoerd-bouma) [inline NuRadioReco/framework/trigger.py:369]: docstring nitpicking - the parameters should be described
```
name: type
   description
```
as elsewhere.

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (sjoerd-bouma) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:167]: There is a built-in functionality to return a bandpass-filtered copy of a channel - `channel.get_filtered_trace(passband, filter_type, order)` ([documentation](https://nu-radio.github.io/NuRadioMC/NuRadioReco/apidoc/NuRadioReco.framework.base_trace.html?highlight=get_filtered_trace#NuRadioReco.framework.base_trace.BaseTrace.get_filtered_trace)). Maybe it doesn't work with a non-default `rp` parameter for the chebyshev filter (yet), but otherwise I think using this function would be a bit tidier?

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (sjoerd-bouma) [inline NuRadioReco/modules/trigger/rnog_surface_trigger.py:101]: docstring nitpickier nitpicking - methods look nicer in sphinx if they have at least a single line description. Would be nice if you could add one for these methods too (can be trivial, e.g. `run the trigger module`).

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (lpyras) [reply NuRadioReco/modules/trigger/rnog_surface_trigger.py:27]: sure, it is temperature now

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (lpyras) [reply NuRadioReco/modules/trigger/rnog_surface_trigger.py:38]: I will remove the definition

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (lpyras) [reply NuRadioReco/modules/trigger/rnog_surface_trigger.py:84]: Thanks! I do that. :)

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (lpyras) [reply NuRadioReco/modules/trigger/rnog_surface_trigger.py:19]: Thanks, I was not aware of that.

File: NuRadioReco/framework/trigger.py
Comment (lpyras) [reply NuRadioReco/framework/trigger.py:369]: Nice one :) I change it.

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (lpyras) [reply NuRadioReco/modules/trigger/rnog_surface_trigger.py:101]: Thanks, I will do that

File: NuRadioReco/modules/trigger/rnog_surface_trigger.py
Comment (lpyras) [reply NuRadioReco/modules/trigger/rnog_surface_trigger.py:167]: Thank you! I didn't know that function. I added the ripple parameter rp. It really looks better now :)

**Review by sjoerd-bouma — APPROVED** [review]
Looks good now! Only minor thing I can think of is I might prefer to have the `calc_parameters_for_rnog_surface_trigger.py` in a subfolder (either the rno_g_surface_trigger_measurements folder, or a `scripts` or `examples` subfolder...) just so you don't eventually end up with a bunch of scripts in the `modules/trigger` module directory.

Feel free to mark conversations as resolved, and you'll have to re-request a review from Anna before merging, but it looks good to me.

**Pull request merged**
